pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -431,6 +436,69 @@ protected CompletableFuture<TransactionPendingAckInternalStats> internalGetPendi
                 );
     }
 
+    protected CompletableFuture<TransactionBufferInternalStats> internalGetTransactionBufferInternalStats(
+            boolean authoritative, boolean metadata) {
+        TransactionBufferInternalStats transactionBufferInternalStats = new TransactionBufferInternalStats();
+        return getExistingPersistentTopicAsync(authoritative)
+                .thenCompose(topic -> {
+                    TransactionBuffer.SnapshotType snapshotType = topic.getTransactionBuffer().getSnapshotType();
+                    if (snapshotType == null) {
+                        return FutureUtil.failedFuture(new RestException(NOT_FOUND,
+                                "Transaction buffer Snapshot for the topic does not exist"));
+                    } else if (snapshotType == TransactionBuffer.SnapshotType.Segment) {
+                        transactionBufferInternalStats.snapshotType = snapshotType.toString();
+                        this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_SEGMENTS);
+                        return getExistingPersistentTopicAsync(authoritative)
```

Comment:
We can not confirm the owner of the topic "__transaction_buffer_snapshot_segments" is the current broker, right? 

If yes, we should get the ml-stats of the topic "__transaction_buffer_snapshot_segments" by another method, it looks like this:

```java
if (currentBrokerIsOwner) {
  //Get ML stats
} else {
  //Send a request to get ML stats
}

```

Reply:
If this is not the owner of the topic, a temporaryRedirect`WebApplicationException(Response.temporaryRedirect(redirect).build())` will be thorwn in  `validateTopicOwnershipAsync`.
So we do not need to take care the `currentBrokerIsOwner`, right?

Reply:
> temporaryRedirectWebApplicationException(Response.temporaryRedirect(redirect).build()) will be thorwn in validateTopicOwnershipAsync. So we do not need to take care the currentBrokerIsOwner, right?

If the error `temporaryRedirectWebApplicationException` is thrown, the HTTP request will always fail. How can users when they get this error?

Reply:
@poorbarcode I found a problem about this. There are multiple topic that needed to get internal managerLedger stats, but it maybe not in the same broker which make we can not use redirect to resolve it. And there are not a API can get all the information in the  `SnapshotInternalStats`.
I have two solutions now:

1.  Add a new API to get topic internal stats.
2. Use the exist API `getInternalStatsAsync` to get `PersistentTopicInternalStats ` which extend `ManagedLedgerInternalStats`. But this way can not get `managedLedgerName`.

```java
 admin.topics().getInternalStatsAsync(topicName.toString(), metadata);
```

I think the `managedLedgerName` is useless, so think we can delete it in the `SnapshotInternalStats` and use the solution2 to resolve this issue.

Reply:
what do you think about it ?